remote_write.go

```diff
@@ -330,6 +389,22 @@ func generateFromTemplates(minValue, maxValue int,
 				},
 			},
 		}
+
+		if debugBuilder != nil {
+			debugBuilder.WriteString(fmt.Sprintf(
+				"series: %s value: %f timestamp: %d\n",
+				labelsToString(series[seriesID-minSeriesID].Labels),
+				series[seriesID-minSeriesID].Samples[0].Value,
+				series[seriesID-minSeriesID].Samples[0].Timestamp,
+			))
+		}
+	}
+
+	if debugWriter != nil && debugBuilder != nil {
+		_, err := debugWriter.Write(debugBuilder.String())
+		if err != nil {
+			log.Fatal(err)
+		}
 	}
```

Comment:
maybe all of this can just be done over the returned `[]prompb.TimeSeries` possibly even not in this function but some of the others where it's also send maybe in `store` itself through a single `dumpTimeSeriesInDebugFile` call